02-oop-inheritance/functions.js

```diff
@@ -0,0 +1,108 @@
+var Movie = function (title, duration, director) {
+	
+		this.title = title;
+		this.duration = duration;
+		this.director = director;
+		
+
+
+};
+
+Movie.prototype = {
+	
+
+
+	play:function(){ MovieObserver.playing(this.title); },
+	stop:function(){ MovieObserver.stopping(this.title); },
+	set:function(attr, value){ this[attr] = value; },
+	get:function(attr){
+		console.log(this[attr]); }
+};
+
+// Observer
+
+var MovieObserver = (function () {
+	var playing = function (name) {
+		console.log("Playing " + name + "...");
+	},
+	stopping = function (name) {
+		console.log("Stopping " + name + "...");
+	};
+
+	return {playing:playing, stopping:stopping};
+
+})();
+
+
+
+// New object + extends
+
+var DownloadableMovie = function (title, duration, director, link){
+	Movie.call(this,title, duration, director);
+	this.link = link;
+
+};
+
+DownloadableMovie.prototype = Object.create(Movie.prototype);
+DownloadableMovie.prototype.getDownloadLink = function (){
+	return "Download Link: " + this.link;
+};
+
+
+var Social= (function () {
+	var share = function (name) {
+		console.log("Sharing " + this.title + " whith " + name);
+	},
+	like = function () {};
+
+	return {share:share, like:like};
+
+})();
+
+function extend(destination, source) {
+  for (var k in source) {
+    if (source.hasOwnProperty(k)) {
+      destination[k] = source[k];
+    }
+  }
+  return destination; 
+}
+
+extend(Movie.prototype, Social);
+extend(DownloadableMovie.prototype, Social);
+
+// Class Actors
+
+
+var Actors= (function () {
```

Comment:
The same here, I don't think an IIFE is need, a simple plain object could do the trick.